Api/Data/QuoteInterface.php

```diff
@@ -45,6 +46,17 @@ public function getPaymentOrderId();
      */
     public function setPaymentOrderId($paymentOrderId);
 
+    /**
+     * @return string|null
+     */
+    public function getPaymentIdPath();
```

Comment:
Can't you just call it `paymentId`? What does the word `path` add here?

```suggestion
    public function getPaymentId();
```

Reply:
It's just method naming. It actually stores the entire `paymentOrderId` ex. `/psp/paymentorders/09ccd29a-7c4f-4752-9396-12100cbfecce`.

Reply:
Then it should be named `PaymentOrderId`, not `PaymentId`, since `paymentorder` and `payment` are different resources in our APIs.

Reply:
We didn't name it `PaymentId`, but rather `PaymentIdPath` :)

Reply:
The word `Payment` is the problem here. It should be `PaymentOrder` since it's not the ID of a `Payment` you are dealing with. You should also not need to handle a parsed GUID and path separately, please make the change in the existing methods to deal with the full URI (path) and detect what has been stored previously to adjust to the full URI.

Reply:
Like I said in other comments here, since there's already a method called `setPaymentOrderId()` that stores the last part of the ID, we have created another method called `setPaymentIdPath()` to store the entire ID & have backwards compatibility. And these methods are only for storing the values in the database.

Reply:
I understand what you have done and why. But you seem to be misunderstanding what I'm trying to express here:

1. In Checkout, you are **not** dealing with **`Payment`** but with **`PaymentOrder`**. You should therefore **not** use **`Payment`** but **`PaymentOrder`** in method names. **`Payment`** and **`PaymentOrder`** are not the same thing.
2. There should be no need to preserve the old methods since our plugin is the only one invoking them. It should be possible to adjust the old methods so they can deal with both UUID (the last part of the URI) and path (full URI) at the same time, no? If you detect the presence of `/` in the ID, it's a path/URI, otherwise it's a UUID and you need to build it up to a full URI. Is this not possible?

Reply:
We can update the method name to `setPaymentOrderIdPath()`. Or we can delete the previous getter-setter methods and replaces them with new ones. 

The reason why we did not refactor the old **PaymentOrderId**, is because it needs some additional work and we prioritised the "phone number validation" issue

Reply:
Ok, then keep the old methods, name the new ones `…paymentOrderIdPath` and then create a task to remove the new ones and refactor the old ones to support both ID formats.

Reply:
All right